user/advanced-topics/usb-qubes.md

```diff
@@ -126,77 +174,122 @@ The first line should read similar to:
+approaches to this problem:
 
 1. Physically disconnect all USB devices whenever you reboot the host.
 2. Hide (i.e., blacklist) all USB controllers from dom0.
 
-**Warning:** If you use a USB [AEM](/doc/anti-evil-maid/) device, do not use the second option.
-Using a USB AEM device requires dom0 to have access to the USB controller to which your USB AEM device is attached.
-If dom0 cannot read your USB AEM device, AEM will hang.
+**Warning:** If you use a USB keyboard, hiding your USB controllers from dom0
+could lock you out of your system. See [USB keyboards](#usb-keyboards) for more
+information.
 
-The procedure to hide all USB controllers from dom0 is as follows:
+**Warning:** Using a USB AEM device requires dom0 to have access to the USB
+controller to which your USB AEM device is attached. If dom0 cannot read your
+USB AEM device, AEM will hang.
 
-* GRUB2
+The following procedure will hide all USB controllers from dom0.
 
-  1. Open the file `/etc/default/grub` in dom0.
-  2. Find the line that begins with `GRUB_CMDLINE_LINUX`.
-  3. Add `rd.qubes.hide_all_usb` to that line.
-  4. Save and close the file.
-  5. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
-  6. Reboot.
+### GRUB2 (legacy boot or EFI)
 
-* EFI
+1. Open the file `/etc/default/grub` in dom0.
+2. Find the line that begins with `GRUB_CMDLINE_LINUX`.
+3. Add `rd.qubes.hide_all_usb` to that line.
+4. Save and close the file.
+5. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` (legacy boot) or
+   `grub2-mkconfig -o /boot/efi/EFI/qubes/grub.cfg` (EFI) in dom0.
+6. Reboot.
 
-  1. Open the file `/boot/efi/EFI/qubes/xen.cfg` in dom0.
-  2. Find the lines that begin with `kernel=`. There may be more than one.
-  3. Add `rd.qubes.hide_all_usb` to those lines.
-  4. Save and close the file.
-  5. Reboot.
 
-## Removing a USB qube ##
+## How to remove a USB qube
 
-**Warning:** This procedure will result in your USB controller(s) being attached directly to dom0.
+**Warning:** This procedure will result in your USB controller(s) being
+attached directly to dom0.
 
-* GRUB2
+### GRUB2
 
-  1. Shut down the USB qube.
-  2. In Qubes Manager, right-click on the USB qube and select "Remove VM."
-  3. Open the file `/etc/default/grub` in dom0.
-  4. Find the line(s) that begins with `GRUB_CMDLINE_LINUX`.
-  5. If `rd.qubes.hide_all_usb` appears anywhere in those lines, remove it.
-  6. Save and close the file.
-  7. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
-  8. Reboot.
+1. Shut down the USB qube.
+2. In Qubes Manager, right-click on the USB qube and select "Remove VM."
+3. Open the file `/etc/default/grub` in dom0.
+4. Find the line(s) that begins with `GRUB_CMDLINE_LINUX`.
+5. If `rd.qubes.hide_all_usb` appears anywhere in those lines, remove it.
+6. Save and close the file.
+7. Run the command `grub2-mkconfig -o /boot/grub2/grub.cfg` in dom0.
+8. Reboot.
 
-* EFI
+### EFI
```

Comment:
Maybe worth clarifying it's about R4.0 only? Because on R4.1 we have Grub on EFI too.